Installer.py

```diff
@@ -1,71 +1,92 @@
     
     # Copies the mods minecraft mods folder
     for mod in os.listdir('./mods'):
-        print('Copying {}...'.format(mod))
+        print(f'{Fore.BLUE}Copying {Fore.YELLOW}{mod}{Fore.WHITE}...')
         shutil.copy('./mods/' + mod, mods_folder)
 
 def installfabric():
-    username = os.getlogin() # The system's username
-    minecraft_folder = f'C:/Users/{username}/AppData/Roaming/.minecraft'
     fabric_folder = minecraft_folder + '/versions/fabric-loader-0.11.6-1.17.1'
 
+    # Detects if the mods folder even exists, if it doesn't, it creates it
+    if not os.path.exists(fabric_folder):
+        print('Fabric folder doesn\'t exist')
+        print(f'{Fore.BLUE}Creating {Fore.YELLOW}fabric folder{Fore.WHITE}...')
+        os.mkdir(fabric_folder)
+        
+        time.sleep(1)
+
     # Deletes the fabric folder and creates a new one if there's already existing fabric is already installed
     if os.listdir(fabric_folder):
-        print('Detected fabric in versions folder')
-        print('Deleting fabric folder...')
+        print('Detected files in fabric folder')
+        print(f'{Fore.RED}Deleting {Fore.YELLOW}fabric folder{Fore.WHITE}...')
 
         shutil.rmtree(fabric_folder, ignore_errors=True)
         
         time.sleep(5) # Added a sleep here, so that if the fabric folder from the minecraft folder is big, this would give it time to actually delete before executing the code below (Change the delay time if you want)
 
-        print('Creating a new fabric folder...')
+        print(f'{Fore.BLUE}Creating {Fore.YELLOW}fabric folder{Fore.WHITE}...')
         os.mkdir(fabric_folder)
     else:
-        print('Detected no mods in mods folder')
+        print('Detected no files in fabric folder')
     
     # Copies the mods minecraft fabric folder
     for mod in os.listdir('./fabric-loader-0.11.6-1.17.1'):
-        print('Copying {}...'.format(mod))
-        shutil.copy('.fabric-loader-0.11.6-1.17.1/' + mod, fabric_folder)
+        print(f'{Fore.BLUE}Copying {Fore.YELLOW}{mod}{Fore.WHITE}...')
+        shutil.copy('./fabric-loader-0.11.6-1.17.1/' + mod, fabric_folder)
 
 if __name__ == "__main__":
     response = input('Start installation? (yes/no): ').lower()
 
     if response == 'yes':
-        print('Installation starting!')
+        print(f'{Fore.GREEN}Installation starting!')
 
-        print('Installing mods!')
+        print(f'{Fore.GREEN}Installing mods!')
         installmods()
-        print('Mods installed!')
+        print(f'{Fore.GREEN}Mods installed!')
 
-        print('Installing fabric!')
+        print(f'{Fore.GREEN}Installing fabric!')
         installfabric()
-        print('Fabric installed!')
+        print(f'{Fore.GREEN}Fabric installed!')
         
-        print('Installation completed!')
-        print('Closing in 20 seconds!')
+        print(f'{Fore.GREEN}Installation completed!')
+        print(f'{Fore.RED}Closing in {Fore.YELLOW}20 seconds{Fore.WHITE}!')
```

Comment:
btw does waiting for 20 seconds for it to close even necessary

Reply:
yes